pkg/stanza/operator/transformer/recombine/recombine.go

```diff
@@ -176,17 +176,17 @@ func (r *Transformer) flushLoop() {
 			timeNow := time.Now()
 			for source, batch := range r.batchMap {
 				entries := batch.entries
-				lastEntryTs := entries[len(entries)-1].ObservedTimestamp
-				timeSinceLastEntry := timeNow.Sub(lastEntryTs)
-				if timeSinceLastEntry < r.forceFlushTimeout {
+				firstEntryTs := entries[0].ObservedTimestamp
+				timeSinceFirstEntry := timeNow.Sub(firstEntryTs)
+				if timeSinceFirstEntry < r.forceFlushTimeout {
 					continue
 				}
 				if err := r.flushSource(source); err != nil {
 					r.Errorf("there was error flushing combined logs %s", err)
 				}
 			}
-
-			r.ticker.Reset(r.forceFlushTimeout)
+			// check every 1/5 forceFlushTimeout
```

Comment:
I see what you're going for here, but it seems like 1/5 is a bit arbitrary.

What do you think about setting the ticker based on `forceFlushTimeout - timeSinceFirstEntry`? If we do this, we should carefully consider if the edge case and perhaps enforce a minimum time until the next check.

Reply:
Yeah, good idea! I've changed the period of the ticker to forceFlushTimeout - maxTimeSinceFirstEntry.

Reply:
@djaglowski  OK, we take the original solution. I will continue to pay attention to this in the future.